modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/DeveloperKeysLayout/Inputs/index.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {ClaySelect} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import {useEffect, useState} from 'react';
+import client from '../../../../../apolloClient';
+import BaseButton from '../../../../../common/components/BaseButton';
+import {useApplicationProvider} from '../../../../../common/context/ApplicationPropertiesProvider';
+import {getListTypeDefinitions} from '../../../../../common/services/liferay/graphql/queries';
+import {fetchDeveloperKeysLicense} from '../../../../../common/services/liferay/raysource-api';
+import {downloadFromBlob} from '../../../../../common/utils';
+import {EXTENSIONS_FILE_TYPE, STATUS_CODE} from '../../../utils/constants';
+
+const DeveloperKeysText = {
+	'DXP':
+		'Select the Liferay DXP version for which you want to download a developer key.',
+	'DXP Cloud':
+		'To activate a local instance of Liferay DXP, download a developer key for your Liferay DXP version.',
+};
+
+const DevelopersKeysInputs = ({
+	accountKey,
+	dxpVersion,
+	productTitle,
+	sessionId,
+}) => {
+	const {licenseKeyDownloadURL} = useApplicationProvider();
+	const [dxpVersions, setDxpVersions] = useState([]);
+
+	const [selectedVersion, setSelectedVersion] = useState(dxpVersion);
+
+	const [hasLicenseDownloadError, setLicenseDownloadError] = useState(false);
+
+	const downloadText = DeveloperKeysText[productTitle];
+
+	useEffect(() => {
+		const fetchListTypeDefinitions = async () => {
+			const {data} = await client.query({
+				query: getListTypeDefinitions,
+				variables: {
+					filter: `name eq 'DXP Version'`,
```

Comment:
Try to pass this 'DXP Version' to a const variable outside the component as `DEVELOPER_KEYS_HELPER`
```suggestion
					filter: `name eq '${LIST_TYPES.dxpVersion}'`,
```

Reply:
Ok